src/charm.py

```diff
@@ -154,13 +177,87 @@ def _on_ingress_revoked(self, event: IngressPerAppRevokedEvent) -> None:
         if self.unit.is_leader():
             logger.info("This app no longer has ingress")
 
+    def _on_openfga_store_created(self, event: OpenFGAStoreCreateEvent) -> None:
+        """Handle openfga store created event."""
+        if not self._peers:
+            self.unit.status = WaitingStatus("Waiting for peer relation")
+            event.defer()
+            return
+
+        if self._get_openfga_store_info():
+            model_id = self._create_openfga_model()
+            self._set_peer_data(key=self._get_version(), data=dict(openfga_model_id=model_id))
+
+            self._handle_status_update_config(event)
+        else:
+            logger.debug("No openfga store info found, deferring the event")
+            event.defer()
+            return
```

Comment:
nitpick: I think it would be cleaner to write this as 
```suggestion
        if not self._get_openfga_store_info():
            logger.debug("No openfga store info found, deferring the event")
            event.defer()
            return

        model_id = self._create_openfga_model()
        self._set_peer_data(key=self._get_version(), data=dict(openfga_model_id=model_id))
        self._handle_status_update_config(event)

```